core/src/components/datetime/datetime.tsx

```diff
@@ -2116,69 +2121,80 @@ export class Datetime implements ComponentInterface {
               dateStyle = getHighlightStyles(highlightedDates, dateIsoString, el);
             }
 
-            return (
-              <button
-                tabindex="-1"
-                data-day={day}
-                data-month={month}
-                data-year={year}
-                data-index={index}
-                data-day-of-week={dayOfWeek}
-                disabled={isCalDayDisabled}
-                class={{
-                  'calendar-day-padding': isCalendarPadding,
-                  'calendar-day': true,
-                  'calendar-day-active': isActive,
-                  'calendar-day-today': isToday,
-                }}
-                style={
-                  dateStyle && {
-                    color: dateStyle.textColor,
-                  }
-                }
-                aria-hidden={isCalendarPadding ? 'true' : null}
-                aria-selected={ariaSelected}
-                aria-label={ariaLabel}
-                onClick={() => {
-                  if (isCalendarPadding) {
-                    return;
-                  }
+            let dateParts = undefined;
 
-                  this.setWorkingParts({
-                    ...this.workingParts,
-                    month,
-                    day,
-                    year,
-                  });
-
-                  // multiple only needs date info, so we can wipe out other fields like time
-                  if (multiple) {
-                    this.setActiveParts(
-                      {
-                        month,
-                        day,
-                        year,
-                      },
-                      isActive
-                    );
-                  } else {
-                    this.setActiveParts({
-                      ...activePart,
+            // Spacer elements should not get the CSS parts added for calendar days
+            if (day !== null) {
```

Comment:
~ nit: Earlier in the implementation we use a variable to track the spacer elements with `isCalendarPadding`, but then we don't use it for two different checks against if the `day` is not null 😆 

Thoughts on doing:
```ts
if (!isCalendarPadding) {
 // existing logic
}
```

Reply:
Sure! Done in https://github.com/ionic-team/ionic-framework/pull/27641/commits/85e250dc950623476aaaf672bf7b76588bc9e2d5

Do you want me to also edit this line: https://github.com/ionic-team/ionic-framework/blob/85e250dc950623476aaaf672bf7b76588bc9e2d5/core/src/components/datetime/datetime.tsx#L2120?